File: 2020/src/main/java/nel/marco/Day1.java
package nel.marco;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;
import java.util.stream.Collectors;

public class Day1 {


    public static List<Integer> readDayOneInput() throws IOException {

        String path = "G:\\Code repositery\\AdventOfCode\\src\\main\\resources\\day1";

        return Files.readAllLines(Path.of(path))
                .stream()
                .map(Integer::valueOf)
                .collect(Collectors.toList());

    }


    public static String partOne() throws IOException {

        List<Integer> input = readDayOneInput();

        for (int i = 0; i < input.size(); i++) {
            for (int j = 1; j < input.size(); j++) {

                Integer first = input.get(i);
                Integer second = input.get(j);
                if (first + second == 2020) {
                    return String.valueOf(first * second);
                }

            }
        }
        return "-1";
    }

    public static String partTwo() throws IOException {

        List<Integer> input = readDayOneInput();

        for (int i = 0; i < input.size(); i++) {
            for (int j = 1; j < input.size(); j++) {
                for (int x = 2; x < input.size(); x++) {

                    Integer first = input.get(i);
                    Integer second = input.get(j);
                    Integer third = input.get(x);

                    if (first + second + third == 2020) {
                        return String.valueOf(first * second * third);
                    }

                }


            }
        }
        return "-1";
    }
}

File: 2019/src/main/kotlin/nel/marco/Day1.kt
package nel.marco

import java.nio.file.Files
import java.nio.file.Path

class Day1 {

    fun part1(input: List<String>): String {

        val answer = input
            .map {
                it.toLong().calculate()
            }
            .sum()

        return answer.toString()
    }

    private fun Long.calculate() = ((this.toDouble() / 3) - 2).toLong()

    private fun calculateRecursively(input: Long): Long {
        if (input <= 3) {
            return 0
        }
        val inputA = input.calculate()
        val InputB = calculateRecursively(inputA)
        return inputA + InputB
    }


    fun part2(input: List<String>): String {
        return input
            .sumOf { calculateRecursively(it.toLong()) }
            .toString()
    }
}

fun main() {
    val lines = Files.readAllLines(Path.of("C:\\code\\AdventOfCode\\2019\\src\\main\\resources\\Day1"))

    println("part 1 = ${Day1().part1(lines)}")
    println("part 2 = ${Day1().part2(lines)}")

}